### PR TITLE
Fix seed executing DDP in aiaccel.torch.apps.train

### PR DESCRIPTION
**Pull request by KanaiYuma-aist**
DDP 実行時に rank 毎に seed が固定化される現象を修正しました。

- ~~関連して、PyTorch Lightning 実行時の rank を参照する環境変数を LOCAL_RANK に修正しました。~~

### CODE REVIEW

File: aiaccel/job/utils/get_rank.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 def get_rank(default: int = 0) -> int:
     for key in [
-        "GLOBAL_RANK",  # PyTorch Lightning
+        "LOCAL_RANK",  # PyTorch Lightning
```
Comment (yoshipon) [inline aiaccel/job/utils/get_rank.py]: こちらやはり危険に見えます．OpenMPI系のコマンドが隠れてしまうので，一旦，Issueにして頂けますか？

**Review by yoshipon — CHANGES_REQUESTED** [review]

File: aiaccel/job/utils/get_rank.py (hunk 1 above)
Comment (KanaiYuma-aist) [reply aiaccel/job/utils/get_rank.py]: issues 作成しました
https://github.com/aistairc/aiaccel/issues/556

File: aiaccel/torch/apps/train.py [hunk 2]
```diff
@@ -37,7 +40,16 @@ def main() -> None:
...
+        if args.seed_distributed:
+            lt.seed_everything(config.seed + rank, workers=True)
+        else:
+            if rank != 0:
+                warnings.warn(
+                    "DDP may be running without '--seed_distributed' option being specified. "
+                    "This feature is planned to be integrated into seed in the future.",
+                    stacklevel=2,
+                )
+            lt.seed_everything(config.seed, workers=True)
```
[the remark this answers is not in the document]
Comment (KanaiYuma-aist) [reply aiaccel/torch/apps/train.py]: 改修しました

- seed_ddp_mode と seed_distributed とで表記ゆれしていたので seed_ddp_mode で統一しました
- 引数 seed_distributed は削除し、config から seed_ddp_mode を読み取って挙動を制御するようにしました

File: aiaccel/job/utils/get_rank.py (hunk 1 above)
Comment (KanaiYuma-aist) [reply aiaccel/job/utils/get_rank.py]: LOCAL_RANK は削除し、 GLOBAL_RANK に戻しています
fd41c5ba1a9e173c470c970bf95d12eb0439b44b

**Comment by KanaiYuma-aist** [comment]
開発者会議メモ
- FutureWarning の内容を反映する（seed_ddp_mode のオプションを削除し、seed_ddp_mode = True の場合の動作をデフォルトの挙動とする）

**Comment by KanaiYuma-aist** [comment]
FutureWarning の削除を行いました。

4994c624c78392925d0492f4e5721304c0da25e1

**Review by yoshipon — APPROVED** [review]
ありがとうございます！
